### ISSUE TITLE
Ifpack2/KokkosKernels? Performance regression with 4.3.1 in AdditiveSchwarz initialization with overlap

### ISSUE THREAD

**Issue by brian-kelley** [trilinos/Trilinos#13013]
Between 45d800 and 53b714, the "remainder" component of preconditioner initialization (AdditiveSchwarz + ILUk, overlap level 1) slowed down significantly on eclipse and amber (for example, 14.2->17.9s on amber for one abnormal energy problem). The remainder is supposed to not include any of the expensive things like RILUK setup, filter construction or overlapping row matrix construction.

**Comment by brian-kelley** [comment]
@csiefer2 Sounds like that should be a separate issue, if it's reproducible and caused by a code change.
Good that we're not measuring that in the stacked timer anymore though.

**Comment by kliegeois** [comment]
@csiefer2 to be precise, the disk read was not in the remainder before #12997.
The PR removed the allocation of one vector (and its initialization to zero), computations of initial norms, conversion of line_info vector to parts arrays, and a block spmv used to make sure that the data moved from host to device.

**Comment by brian-kelley** [comment]
@csiefer2 @kliegeois I think i figured both of these things out (but they're not related)
- This issue appears to be 100% due to software environment changes on amber and eclipse. On amber we had to explicitly chang the compiler we use to intel 24 because intel 21 went away, and this happened on the exact day of the spike. Not sure about the details on eclipse. I wasn't able to replicate the old "good" time on either machine. Both the good and bad SHAs above with the current environment give the "bad" time on both machines. 
- The sparc single_gpu remainder slowdown was due to cusparse BSR spmv being extremely slow on the first call. I'm talking to NVIDIA about what we could do about it. Just using our native impl in that case seems like a reasonable workaround though.
